### PR TITLE
Update job name tests for hard failure cases

### PR DESCRIPTION
**Pull request by stbenjam**
[TRT-470](https://issues.redhat.com//browse/TRT-470)

- Hard fail if a job name indicates a network type, but job is not using
  it. For example, a job with sdn in the name that's really ovn.
- Hard fail if any periodic is missing the network type
- Flake if we see a network provider that's not SDN or OVN.  Most code
  in origin assumes it's either/or but there's a request from the
  network folks that we might have a few jobs using other providers.
  This should help find them.

### CODE REVIEW

File: test/extended/ci/job_names.go [hunk 1]
```diff
@@ -70,15 +72,33 @@ var _ = g.Describe("[sig-ci] [Early] prow job name", func() {
 
 		network, err := oc.AdminConfigClient().ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
 		o.Expect(err).NotTo(o.HaveOccurred())
+		isPeriodic := strings.HasPrefix(jobName, "periodic-")
 		switch network.Status.NetworkType {
 		case "OpenShiftSDN":
-			if !strings.Contains(jobName, "sdn") {
-				result.Flakef("job name %q does not have network type in name (expected `sdn`)", jobName)
```
[the remark this answers is not in the document]
Comment (DennisPeriquet) [reply test/extended/ci/job_names.go:76]: hm... if a flake can change to a fail (if line 79 is run), then the logic works.

[the remark this answers is not in the document]
Comment (DennisPeriquet) [reply test/extended/ci/job_names.go:76]: ignore me.  That's a deleted line.

[the remark this answers is not in the document]
Comment (DennisPeriquet) [reply test/extended/ci/job_names.go:76]: The logic makes sense.

**Comment by stbenjam** [comment]
@DennisPeriquet Thanks! Sorry I missed you had one open already, but they look almost the same.  Thanks!

**Pull request merged**